### PR TITLE
Better document process how types are generated.

### PR DESCRIPTION
**Pull request by christian-bromann**
**Pre-check**
- [x] I'm aware that I can [edit the docs](https://github.com/webdriverio/webdriverio/blob/master/CONTRIBUTING.md) and submit a pull request

**Describe the improvement**

I'd like to report
- [ ] Unclear documentation
- [ ] A typo
- [x] Missing documentation
- [ ] Other

**Description of the improvement / report**
We should better document how types are generated, so we don't need to explain this process [during PR reviews](https://github.com/webdriverio/webdriverio/pull/4946#discussion_r364488285).

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -94,6 +94,107 @@ cd your-main-test-code
...
+This will run the scripts in the directory shown above and generate the typings various packages located in `/packages/*`. Each package defines its typings differently:
+
+- the `webdriver` package defines all commands and their respective types in the json file of the `@wdio/protocols` package
+- the `webdriverio` package has the types defined in the comments above every command in `/packages/webdriverio/src/commands/**/*`
+- the `@wdio/allure-reporter` has its interface defined in comments within the service (`/packages/wdio-allure-reporter/src/index.js`)
+- all other packages have their type definitions directly in the `*.d.ts` in the root directory of that package (e.g. `/packages/wdio-cucumber-framework/cucumber-framework.d.ts`)
+
+Whenever you change the types in these packages, make sure you re-generate the types with the command shown above. In order to ensure that packages where type definitions have changed are released we keep the generated type definition in source control.
+
+### Adding Types for webdriverIO
```
Comment (mgrybyk) [inline CONTRIBUTING.md]: should WebdriverIO start from uppercase here?

**Comment by mgrybyk** [comment]
Maybe we should also mention that types should be generated if there were any changes to js doc or protocols?

File: CONTRIBUTING.md (hunk 1 above)
Comment (christian-bromann) [reply CONTRIBUTING.md]: 👍

**Comment by christian-bromann** [comment]
> Maybe we should also mention that types should be generated if there were any changes to js doc or protocols?

I thought I covered this by 

> Whenever you change the types in these packages, make sure you re-generate the types with the command shown above.

**Pull request merged**